specification/communication/data-plane/Microsoft.CommunicationServicesChat/preview/2021-03-01-preview5/communicationserviceschat.json

```diff
@@ -0,0 +1,1690 @@
+        },
+        "x-ms-examples": {
+          "Create chat thread": {
+            "$ref": "./examples/Threads_CreateChatThread.json"
+          },
+          "Create chat thread with repeatability request id header": {
+            "$ref": "./examples/Threads_CreateChatThreadWithIdempotencyToken.json"
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "Threads"
+        ],
+        "summary": "Gets the list of chat threads of a user.",
+        "operationId": "Chat_ListChatThreads",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "query",
+            "name": "maxPageSize",
+            "description": "The maximum number of chat threads returned per page.",
+            "type": "integer",
+            "format": "int32"
+          },
+          {
+            "in": "query",
+            "name": "startTime",
+            "description": "The earliest point in time to get chat threads up to. The timestamp should be in RFC3339 format: `yyyy-MM-ddTHH:mm:ssZ`.",
+            "type": "string",
+            "format": "date-time"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The action returns a `GetThreadsResponse` resource.",
+            "schema": {
+              "$ref": "#/definitions/ChatThreadsItemCollection"
+            }
+          },
+          "401": {
+            "description": "Unauthorized.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
+          "403": {
+            "description": "Forbidden.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
+          "429": {
+            "description": "Too many requests.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
+          "503": {
+            "description": "Service unavailable.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          }
+        },
+        "x-ms-examples": {
+          "Get threads with pagination (Max Page Size)": {
+            "$ref": "./examples/Threads_ListChatThreadsWithPageSize.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink",
+          "itemName": "value"
+        }
+      }
+    },
+    "/chat/threads/{chatThreadId}": {
+      "patch": {
+        "tags": [
+          "Threads"
+        ],
+        "summary": "Updates a thread's properties.",
+        "operationId": "ChatThread_UpdateChatThreadProperties",
```

Comment:
The method name here, UpdateChatThreadProperties, is a bit redundant.  The way this works for SDK codegen is that the value to the left of the `_` defines a class/interface and the value to the right defines a method.  Since the method is on a class named `ChatThread`, having "ChatThread" in the method name seems redundant.  Not a blocking issue, please ack if you're ok with the name.  Same comment applies to `ChatThread_GetChatThreadProperties` below.

Reply:
We're generating 2 clients a main `Chat` client and a nested `ChatThread` client, by adding `ChatThread` prefix for the `UpdateChatThreadProperties` we are defining which client would have the generated code for the method, in this case the `ChatThread` client.

Reply:
Yes I understand that, I'm referring to the method name (the client names are great).

Reply:
```C#
class ChatThread {
  public void UpdateChatThreadProperties()
}
```
I'm saying the "ChatThread" substring in the method name is redundant.

Reply:
I see your point. 

The reason for using a fully qualified name is to avoid ambiguity with other update methods that are defined on the `ChatThread` client. Currently, we have `ChatThread_UpdateChatMessage` as well, and we don't to have any ambiguity in the API now, or if we need to add more methods in the future.